drivers/bluetooth/hci/spi.c

```diff
@@ -480,26 +482,21 @@ static int bt_spi_open(void)
 {
 	/* Configure RST pin and hold BLE in Reset */
 	gpio_pin_configure(rst_dev, GPIO_RESET_PIN,
-			   GPIO_DIR_OUT | GPIO_PUD_PULL_UP);
-	gpio_pin_write(rst_dev, GPIO_RESET_PIN, 0);
+			   GPIO_OUTPUT | GPIO_RESET_FLAGS);
+	gpio_pin_set(rst_dev, GPIO_RESET_PIN, 1);
 
 	/* Configure IRQ pin and the IRQ call-back/handler */
 	gpio_pin_configure(irq_dev, GPIO_IRQ_PIN,
-#if defined(CONFIG_BT_SPI_BLUENRG)
-			   GPIO_PUD_PULL_DOWN |
-#endif
-			   GPIO_DIR_IN | GPIO_INT |
-			   GPIO_INT_EDGE | GPIO_INT_ACTIVE_HIGH);
+			   GPIO_INPUT | GPIO_IRQ_FLAGS);
 
 	gpio_init_callback(&gpio_cb, bt_spi_isr, BIT(GPIO_IRQ_PIN));
 
 	if (gpio_add_callback(irq_dev, &gpio_cb)) {
 		return -EINVAL;
 	}
 
-	if (gpio_pin_enable_callback(irq_dev, GPIO_IRQ_PIN)) {
-		return -EINVAL;
-	}
+	gpio_pin_interrupt_configure(irq_dev, GPIO_IRQ_PIN,
+				     GPIO_INT_EDGE_RISING);
```

Comment:
Since we want to support logical levels (honor `GPIO_ACTIVE_LOW`, `GPIO_ACTIVE_HIGH` flags) we should use here `GPIO_INT_EDGE_TO_ACTIVE` flag.